src/components/TaskList/TaskList.jsx

```diff
@@ -9,68 +9,103 @@ const TaskVisibility = require('../../models/TaskVisibility')
 class TaskList extends React.Component {
   constructor(props) {
     super(props)
-    this.state = { selectedIndex: null }
-    this.onKeyUp = this.onKeyUp.bind(this)
-    this.onKeyDown = this.onKeyDown.bind(this)
+    this.state = { focusedIndex: 0, showHidden: false }
   }
 
   componentDidMount() {
-    document.addEventListener('keyup', this.onKeyUp)
-    document.addEventListener('keydown', this.onKeyDown)
+    this.keyDownEventListener = event => this.onKeyDown(event)
+    document.addEventListener('keydown', this.keyDownEventListener)
   }
 
   componentWillUnmount() {
-    document.removeEventListener('keyup', this.onKeyUp)
-    document.removeEventListener('keydown', this.onKeyDown)
+    document.removeEventListener('keydown', this.keyDownEventListener)
   }
 
-  onSnoozeClick(event) {
-    event.currentTarget.blur() // defocus button
-    this.props.dispatch({ type: 'TASKS_SNOOZE' })
+  onKeyDown(event) {
+    const up = ['ArrowUp', 'k']
+    const down = ['ArrowDown', 'j']
+    const open = ['Enter', 'o']
+    const select = [' ', 'x']
+    const escape = ['Escape']
+    const archive = ['a']
+    const snooze = ['s']
+    const ignore = ['i']
+
+    let preventDefault = true
+
+    if (up.includes(event.key)) {
+      this.focusPreviousTask()
+    } else if (down.includes(event.key)) {
+      this.focusNextTask()
+    } else if (escape.includes(event.key)) {
+      this.setState({ focusedIndex: 0 })
+    } else if (open.includes(event.key)) {
+      this.openFocusedTask()
+    } else if (select.includes(event.key)) {
+      this.onToggle()
+    } else if (snooze.includes(event.key)) {
+      this.snooze()
+    } else if (archive.includes(event.key)) {
+      this.archive()
+    } else if (ignore.includes(event.key)) {
+      this.ignore()
+    } else {
+      preventDefault = false
+    }
+
+    if (preventDefault) event.preventDefault()
   }
 
-  onArchiveClick(event) {
-    event.currentTarget.blur() // defocus button
-    this.props.dispatch({ type: 'TASKS_ARCHIVE' })
+  tasks() {
```

Comment:
Is this a method as opposed to filtering and passing only the applicable tasks as part of the properties list to the component so that we're able to change which tasks are visible on the fly? You couldn't re-render the whole component with new properties when hidden/visible changes? I don't even know if that's possible, just wondering.